bigquery/bigquery_storage_quickstart/main.go

```diff
@@ -266,7 +314,49 @@ func processStream(ctx context.Context, client *bqStorage.BigQueryReadClient, st
 				offset = offset + rc
 				// We're making progress, reset retries.
 				retries = 0
-				ch <- r.GetAvroRows()
+				ch <- r
+			}
+		}
+	}
+}
+
+// processArrow receives row blocks from a channel, and uses the provided Arrow
+// schema to decode the blocks into individual row messages for printing.  Will
+// continue to run until the channel is closed or the provided context is
+// cancelled.
+func processArrow(ctx context.Context, schema []byte, ch <-chan *bqStoragepb.ReadRowsResponse) error {
+	mem := memory.NewGoAllocator()
+	buf := bytes.NewBuffer(schema)
+	r, err := ipc.NewReader(buf, ipc.WithAllocator(mem))
+	if err != nil {
+		return err
+	}
+	aschema := r.Schema()
+	for {
+		select {
+		case <-ctx.Done():
+			// Context was cancelled.  Stop.
+			return nil
```

Comment:
nit: This error should be propagated back to the user.

Reply:
Does returning `ctx.Err()` make sense in this context ? 

```
case <-ctx.Done():
	// Context was cancelled.  Stop.
	return ctx.Err()
```